include/picongpu/plugins/radiation/Radiation.hpp

```diff
@@ -664,396 +659,197 @@ namespace picongpu
+                    // write mesh attributes
+                    mesh_n.setGeometry(::openPMD::Mesh::Geometry::cartesian); // set be default
+                    // mesh_n.setGeometryParameters( geometryParameters );
+                    mesh_n.setDataOrder(::openPMD::Mesh::DataOrder::C);
+                    mesh_n.setGridSpacing(std::vector<double>{1.0, 1.0, 1.0});
+                    mesh_n.setGridGlobalOffset(std::vector<double>{0.0, 0.0, 0.0});
+                    mesh_n.setGridUnitSI(1.0);
+                    mesh_n.setAxisLabels(std::vector<std::string>{"detector direction index", "None", "None"});
+                    mesh_n.setUnitDimension(std::map<::openPMD::UnitDimension, double>{});
 
+                    {
+                        std::vector<float_64> tmpBuffer(parameters::N_observer);
 
-                    /* save detector frequencies */
-                    splash::Dimensions bufferSizeOmega(1, radiation_frequencies::N_omega, 1);
+                        // reshape abstract MeshRecordComponent
+                        ::openPMD::Datatype datatype_n = ::openPMD::determineDatatype(::openPMD::shareRaw(tmpBuffer));
+                        ::openPMD::Extent extent_n = {parameters::N_observer, 1, 1};
+                        ::openPMD::Offset offset_n = {0, 0, 0};
 
-                    splash::Dimensions strideOmega(1, 1, 1);
+                        const picongpu::float_64 factorDirection = 1.0;
 
-                    splash::Dimensions offset(0, 0, 0);
-                    splash::Selection dataSelection(bufferSizeOmega, bufferSizeOmega, offset, strideOmega);
+                        for(uint32_t detectorDim = 0; detectorDim < 3; ++detectorDim)
+                        {
+                            std::string dir = dataLabelsDetectorDirection(detectorDim);
+                            mesh_n[dir].setUnitSI(factorDirection);
+                            mesh_n[dir].setPosition(std::vector<double>{0.0, 0.0, 0.0});
+                            ::openPMD::Dataset dataset_n = ::openPMD::Dataset(datatype_n, extent_n);
+                            mesh_n[dir].resetDataset(dataset_n);
+
+                            // select data
+                            for(int copyIndex = 0; copyIndex < parameters::N_observer; ++copyIndex)
+                            {
+                                tmpBuffer[copyIndex]
+                                    = ((picongpu::float_64*) detectorPositions.data())[detectorDim + 3 * copyIndex];
+                            }
 
-                    hdf5DataFile.write(
-                        currentStep,
-                        radSplashType,
-                        3,
-                        dataSelection,
-                        (meshesPathName + dataLabelsDetectorFrequency(0)).c_str(),
-                        detectorFrequencies.data());
+                            // write actual data
+                            mesh_n[dir].storeChunk(::openPMD::shareRaw(tmpBuffer), offset_n, extent_n);
```

Comment:
Same as above. No explicit `shareRaw` is needed here, this might benefit from using the Span-based `storeChunk`-API.